File: backend/app/schemas/goal.py
from datetime import datetime, date
from typing import Optional, Literal
from uuid import UUID
from pydantic import BaseModel


GoalStatus = Literal["active", "completed", "paused", "abandoned"]
JournalingSchedule = Literal["morning", "evening", "both"]


class GoalCreate(BaseModel):
    title: str
    description: Optional[str] = None
    target_date: Optional[date] = None
    journaling_schedule: Optional[JournalingSchedule] = None


class GoalUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[GoalStatus] = None
    target_date: Optional[date] = None
    journaling_schedule: Optional[JournalingSchedule] = None


class GoalResponse(BaseModel):
    id: UUID
    title: str
    description: Optional[str]
    status: str
    target_date: Optional[date]
    journaling_schedule: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: ios/JournalBuddy/JournalBuddy/Models/Goal.swift
import Foundation

enum GoalStatus: String, Codable, CaseIterable {
    case active
    case completed
    case paused
    case abandoned

    var displayName: String {
        rawValue.capitalized
    }

    var icon: String {
        switch self {
        case .active: return "target"
        case .completed: return "checkmark.circle.fill"
        case .paused: return "pause.circle"
        case .abandoned: return "xmark.circle"
        }
    }
}

enum JournalingSchedule: String, Codable, CaseIterable {
    case morning
    case evening
    case both

    var displayName: String {
        switch self {
        case .morning: return "Morning"
        case .evening: return "Evening"
        case .both: return "Both"
        }
    }
}

struct Goal: Codable, Identifiable {
    let id: UUID
    var title: String
    var description: String?
    var status: GoalStatus
    var progress: Int
    var targetDate: Date?
    var journalingSchedule: JournalingSchedule?
    let createdAt: Date
    let updatedAt: Date

    enum CodingKeys: String, CodingKey {
        case id, title, description, status, progress
        case targetDate = "target_date"
        case journalingSchedule = "journaling_schedule"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

struct GoalCreate: Codable {
    var title: String
    var description: String?
    var targetDate: Date?
    var journalingSchedule: JournalingSchedule?

    enum CodingKeys: String, CodingKey {
        case title, description
        case targetDate = "target_date"
        case journalingSchedule = "journaling_schedule"
    }
}

struct GoalUpdate: Codable {
    var title: String?
    var description: String?
    var status: GoalStatus?
    var progress: Int?
    var targetDate: Date?
    var journalingSchedule: JournalingSchedule?

    enum CodingKeys: String, CodingKey {
        case title, description, status, progress
        case targetDate = "target_date"
        case journalingSchedule = "journaling_schedule"
    }
}
